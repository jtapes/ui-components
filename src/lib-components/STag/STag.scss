@import '../../theme/styles/variables';
@import '../../theme/styles/mixins';
@import '../../theme/styles/reset';
@import '../../theme/styles/fonts';
@import '../../theme/styles/global';

.s-tag {
  position: relative;
  display: inline-flex;
  padding: $size--8 $size--16;

  &::before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: $color--white;
    opacity: 0.92;
  }

  &__content {
    @include z-index(default);

    white-space: nowrap;
  }

  &--filled {
    &::before {
      visibility: hidden;
    }
  }

  & + & {
    margin-left: $size--4;
  }
}
