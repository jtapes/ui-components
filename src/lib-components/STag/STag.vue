<template>
  <div ref="tag" class="s-tag text-sm" :class="tagClasses">
    <span class="s-tag__content">
      <!-- @slot Слот для контента тега  -->
      <slot>
        {{ label }}
      </slot>
    </span>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

/** Компонент для отображения дополнительной информации */
@Component
export default class STag extends Vue {
  /** Цвет текста (наименование) */
  @Prop(String)
  textColor!: string

  /** Цвет фона (наименование) */
  @Prop(String)
  bgColor?: string

  /** Контент тега */
  @Prop(String)
  label!: string

  get tagClasses(): (string | null)[] {
    const color = `color-${this.textColor}`
    let backgroundColor = `bg-${this.textColor}`
    let filled = null

    if (this.bgColor) {
      backgroundColor = `bg-${this.bgColor}`
      filled = 's-tag--filled'
    }

    return [color, backgroundColor, filled]
  }
}
</script>
