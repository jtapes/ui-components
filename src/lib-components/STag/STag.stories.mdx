import { action } from '@storybook/addon-actions'
import { config, withDesign } from 'storybook-addon-designs'
import { withTests } from '@storybook/addon-jest';
import {
  Meta,
  Story,
  ArgsTable,
  Description,
  Source,
  Props,
  Preview,
} from '@storybook/addon-docs'
import results from '/.jest-test-results.json';
import STag from '@/lib-components/STag/STag.vue'

<Meta
  title="DESIGN SYSTEM/Atom/Tag"
  component={STag}
  decorators={[withDesign, withTests({results})]}
  argTypes={{
    bgColor: { control: 'text' },
    textColor: { control: 'text' },
    label: { control: 'text' },
  }}
  parameters={{
    design: config({
      type: 'figma',
      url:
        'https://www.figma.com/file/42Xh2qI3qvq2CRkc2HJlwA/MVP-3-Market-Catalog-%E2%80%A2-Product-Page?node-id=1128%3A88',
    }),
    docs: {
      source: {
        type: 'code',
      },
    },
  }}
/>

# Тег

<Description of={STag} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { STag },
  template:
    `<s-tag :bgColor="bgColor" :textColor="textColor" :label="label" />`,
})

<Preview withToolbar withSource="none">
  <Story name="STag" args={{ textColor: 'diamond', label: 'Возможна резка' }}>
    {Template.bind({})}
  </Story>
</Preview>

## Пример

<Source
  language="html"
  code={`
    <s-tag bg-color="diamond-light" text-color="diamond" label="Возможна резка" />
  `}
/>

<ArgsTable of={STag} />

<div style={{padding: '10px 0', marginBottom: '15px'}}>
    <span style={{color: '#111111', fontWeight: 700, fontSize: '20px'}}>Все значения цветов пишутся классом, если не указывать цвет фона, то он автоматически выставится как цвет текста с прозрачностью 0.08 </span>
</div>
