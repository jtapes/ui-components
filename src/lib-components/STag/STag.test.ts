import { mount } from '@vue/test-utils'
import STag from '@/lib-components/STag/STag.vue'

describe('STag', () => {
  test('Is Vue snapshot', () => {
    const wrapper = mount(STag, {
      propsData: {
        label: '1213',
        bgColor: 'primary',
        textColor: 'primary-accent',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
