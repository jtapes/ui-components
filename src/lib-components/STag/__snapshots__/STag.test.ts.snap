// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STag Is Vue snapshot 1`] = `
<div
  class="s-tag text-sm color-primary-accent bg-primary s-tag--filled"
>
  <span
    class="s-tag__content"
  >
    
      1213
    
  </span>
</div>
`;
